### ISSUE TITLE
Fail to decode transaction correctly.

### ISSUE THREAD

**Issue by JiechengZhao** [bitcoin/bitcoin#11157]
### Describe the issue
decode to a strange result of a raw transaction 

#### If so, please list the steps to reproduce below:
use `bitcoin-cli decoderawtransaction 010000000001011d35c185f8bb0fe4d6f4ccfc06412040bcdce7e67614990d2eae82dc9cc5e6620000000017160014750849d2d607b9a5085a78b2a4daa5404ac3c4c2ffffffff02102700000000000017a914f7cd0db841b1da5c613194f16924ef5a3c800b1c87d354d0000000000017a914f7ef5bb0d096f5b96b7ad6d4c999d66a0776f33d870247304402203c424595c2ad559f33769d494eefedfa71d3312cb1e6fb0f64fbdbe358aaf6d602204f26789a64457e972c705757023978ed825e45a051ea07744773c6899e5bfbcd012102fa03fe77aa01b00bff222732de668f12084231e29a3dffbbd62322889afcf4a800000000`

The transaction id is `89c86c5dddb4786324df8cb5a4d67537b7cf4b800e4432e3c7d60f38adb7046a`

### Expected behaviour
as https://blockchain.info/decode-tx show, it should be:
```
{
   "lock_time":0,
   "size":247,
   "inputs":[
      {
         "prev_out":{
            "index":0,
            "hash":"62e6c59cdc82ae2e0d991476e6e7dcbc40204106fcccf4d6e40fbbf885c1351d"
         },
         "script":"160014750849d2d607b9a5085a78b2a4daa5404ac3c4c2"
      }
   ],
   "version":1,
   "vin_sz":1,
   "hash":"89c86c5dddb4786324df8cb5a4d67537b7cf4b800e4432e3c7d60f38adb7046a",
   "vout_sz":2,
   "out":[
      {
         "script_string":"OP_HASH160 f7cd0db841b1da5c613194f16924ef5a3c800b1c OP_EQUAL",
         "address":"3QHGUJJzHiq2FFUnJF8MJayp6ZpmwHDKY6",
         "value":10000,
         "script":"a914f7cd0db841b1da5c613194f16924ef5a3c800b1c87"
      },
      {
         "script_string":"OP_HASH160 f7ef5bb0d096f5b96b7ad6d4c999d66a0776f33d OP_EQUAL",
         "address":"3QHyZpgKmm39371utHvQMM8WNFTRjFa7NU",
         "value":13653203,
         "script":"a914f7ef5bb0d096f5b96b7ad6d4c999d66a0776f33d87"
      }
   ]
}
```
### Actual behaviour
A very large vout value is included `11337729845.69634049` which definitely is wrong.
```
{
  "txid": "96a5f667ab421ce2ae92d3f54727a191cd1a95a55a10efa3817e93534f725038",
  "hash": "96a5f667ab421ce2ae92d3f54727a191cd1a95a55a10efa3817e93534f725038",
  "size": 247,
  "vsize": 247,
  "version": 1,
  "locktime": 0,
  "vin": [
  ],
  "vout": [
    {
      "value": 11337729845.69634049,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_UNKNOWN OP_UNKNOWN OP_UNKNOWN OP_UNKNOWN 412040bcdce7 OP_UNKNOWN OP_DUP 990d2eae82dc9cc5e66200000000171600147508 d2d607b9a5085a78b2a4daa5404ac3c4c2ffffffff02102700000000000017a914f7cd0db841b1da5c613194f16924ef5a3c800b1c87d354d0000000000017a914f7ef5bb0d096f5b9 OP_TOALTSTACK OP_ROLL OP_UNKNOWN OP_UNKNOWN OP_UNKNOWN OP_RSHIFT OP_UNKNOWN OP_RETURN 76f33d87024730 02203c424595c2ad559f33769d494eefedfa71d3312cb1e6fb0f64fbdbe358aaf6d602204f26789a64457e972c705757023978ed825e45a051ea07744773c6899e5bfbcd 33 1018 OP_UNKNOWN OP_NIP OP_HASH256 -48 ff222732de668f12084231 OP_UNKNOWN OP_BOOLAND [error]",
        "hex": "d6f4ccfc06412040bcdce7e67614990d2eae82dc9cc5e6620000000017160014750849d2d607b9a5085a78b2a4daa5404ac3c4c2ffffffff02102700000000000017a914f7cd0db841b1da5c613194f16924ef5a3c800b1c87d354d0000000000017a914f7ef5bb0d096f5b96b7ad6d4c999d66a0776f33d870247304402203c424595c2ad559f33769d494eefedfa71d3312cb1e6fb0f64fbdbe358aaf6d602204f26789a64457e972c705757023978ed825e45a051ea07744773c6899e5bfbcd012102fa03fe77aa01b00bff222732de668f12084231e29a3dffbbd62322889afcf4a8",
        "type": "nonstandard"
      }
    }
  ]
}
```

### What version of bitcoin-core are you using?
version 0.14.2 
git commit fc61c8322bd7288f7546d18ad04c36c345be13cd , self compiled.

### Machine specs:
- OS: MacOS Darwin 16.7.0 Darwin Kernel Version 16.7.0: Thu Jun 15 17:36:27 PDT 2017; root:xnu-3789.70.16~2/RELEASE_X86_64 x86_64
- CPU:   Model Name:	MacBook Pro
  Processor Speed:	2.5 GHz
  L2 Cache (per Core):	256 KB
  L3 Cache:	6 MB
- RAM: 
  Memory:	16 GB
- Disk size:
  500G
- Disk Type (HD/SDD):
  SDD

### Any extra information that might be useful in the debugging process.
This is normally the contents of a `debug.log` or `config.log` file. Raw text or a link to a pastebin type site are preferred.
```
(venv) zjc@MP:bt-py$ bitcoin-cli getrawtransaction 89c86c5dddb4786324df8cb5a4d67537b7cf4b800e4432e3c7d60f38adb7046a

010000000001011d35c185f8bb0fe4d6f4ccfc06412040bcdce7e67614990d2eae82dc9cc5e6620000000017160014750849d2d607b9a5085a78b2a4daa5404ac3c4c2ffffffff02102700000000000017a914f7cd0db841b1da5c613194f16924ef5a3c800b1c87d354d0000000000017a914f7ef5bb0d096f5b96b7ad6d4c999d66a0776f33d870247304402203c424595c2ad559f33769d494eefedfa71d3312cb1e6fb0f64fbdbe358aaf6d602204f26789a64457e972c705757023978ed825e45a051ea07744773c6899e5bfbcd012102fa03fe77aa01b00bff222732de668f12084231e29a3dffbbd62322889afcf4a800000000

(venv) zjc@MP:bt-py$ bitcoin-cli decoderawtransaction 010000000001011d35c185f8bb0fe4d6f4ccfc06412040bcdce7e67614990d2eae82dc9cc5e6620000000017160014750849d2d607b9a5085a78b2a4daa5404ac3c4c2ffffffff02102700000000000017a914f7cd0db841b1da5c613194f16924ef5a3c800b1c87d354d0000000000017a914f7ef5bb0d096f5b96b7ad6d4c999d66a0776f33d870247304402203c424595c2ad559f33769d494eefedfa71d3312cb1e6fb0f64fbdbe358aaf6d602204f26789a64457e972c705757023978ed825e45a051ea07744773c6899e5bfbcd012102fa03fe77aa01b00bff222732de668f12084231e29a3dffbbd62322889afcf4a800000000

{
  "txid": "96a5f667ab421ce2ae92d3f54727a191cd1a95a55a10efa3817e93534f725038",
  "hash": "96a5f667ab421ce2ae92d3f54727a191cd1a95a55a10efa3817e93534f725038",
  "size": 247,
  "vsize": 247,
  "version": 1,
  "locktime": 0,
  "vin": [
  ],
  "vout": [
    {
      "value": 11337729845.69634049,
      "n": 0,
      "scriptPubKey": {
        "asm": "OP_UNKNOWN OP_UNKNOWN OP_UNKNOWN OP_UNKNOWN 412040bcdce7 OP_UNKNOWN OP_DUP 990d2eae82dc9cc5e66200000000171600147508 d2d607b9a5085a78b2a4daa5404ac3c4c2ffffffff02102700000000000017a914f7cd0db841b1da5c613194f16924ef5a3c800b1c87d354d0000000000017a914f7ef5bb0d096f5b9 OP_TOALTSTACK OP_ROLL OP_UNKNOWN OP_UNKNOWN OP_UNKNOWN OP_RSHIFT OP_UNKNOWN OP_RETURN 76f33d87024730 02203c424595c2ad559f33769d494eefedfa71d3312cb1e6fb0f64fbdbe358aaf6d602204f26789a64457e972c705757023978ed825e45a051ea07744773c6899e5bfbcd 33 1018 OP_UNKNOWN OP_NIP OP_HASH256 -48 ff222732de668f12084231 OP_UNKNOWN OP_BOOLAND [error]",
        "hex": "d6f4ccfc06412040bcdce7e67614990d2eae82dc9cc5e6620000000017160014750849d2d607b9a5085a78b2a4daa5404ac3c4c2ffffffff02102700000000000017a914f7cd0db841b1da5c613194f16924ef5a3c800b1c87d354d0000000000017a914f7ef5bb0d096f5b96b7ad6d4c999d66a0776f33d870247304402203c424595c2ad559f33769d494eefedfa71d3312cb1e6fb0f64fbdbe358aaf6d602204f26789a64457e972c705757023978ed825e45a051ea07744773c6899e5bfbcd012102fa03fe77aa01b00bff222732de668f12084231e29a3dffbbd62322889afcf4a8",
        "type": "nonstandard"
      }
    }
  ]
}
```

**Comment by MeshCollider** [comment]
That transaction is a witness transaction, but 0.14.2 is interpreting it as non-witness, which is why the hash is the same as the txid and there are no inputs. 0.15.0 interprets it fine.

**Comment by achow101** [comment]
This should be fixed in 0.15.0 by #10481

**Comment by JiechengZhao** [comment]
Thank you.
